src/test/java/com/mmm/his/cer/utility/farser/ast_complex/setup/ComplexTestTokenType.java

```diff
@@ -19,6 +19,8 @@ public enum ComplexTestTokenType implements AstTokenType<ComplexTestTokenType> {
   LPAREN("(", AstCommonTokenType.LPAREN),
   RPAREN(")", AstCommonTokenType.RPAREN),
   NOT("!", AstCommonTokenType.NOT),
+  FUNCTION(".", AstCommonTokenType.FUNCTION),
```

Comment:
Yeah I added the function symbol to make it simple. If we changed the token iterator to instead be a LinkedList it could probably stay pretty simple as that has a built in peek method. Otherwise we would have to keep track of if we already looked forward or not before moving the token forward, which is also doable. I see the `LtrExpressionIterator` does that

Reply:
Update- got rid of the function call so now it is just an atom followed by a LPAREN

Reply:
Nice!
I see your change from an `Iterator` to a `List`. The user will not know that a `LinkedList` could get passed in. Could you document that in the javadoc or provide overloaded methods which accepts a `LinkedList`?

Have you guys discussed the overhead of creating a linked list each time from a list? Probably minimal... but a custom lookahead (custom iterator implementation) could solve that.

Reply:
The lexer now returns a linked list, so most likely the list will be able to just get cast to a linked list [here](https://github.com/solventum-oss/farser/blob/functions/src/main/java/com/mmm/his/cer/utility/farser/ast/parser/AstDescentParser.java#L62) and no conversion will need to happen. It probably makes sense to just have it accept a LinkedList though and not a generic list

Reply:
Ok, good to know.

Generic list is good as it is, so that you do not force users to use a linked list. Linked list would be ok as overloaded method input.